internal/operator-controller/authorization/rbac.go

```diff
@@ -58,59 +63,61 @@ func NewRBACPreAuthorizer(cl client.Client) PreAuthorizer {
 }
 
 // PreAuthorize validates whether the current user/request satisfies the necessary permissions
-// as defined by the RBAC policy. It examines the user’s roles, resource identifiers, and
-// the intended action to determine if the operation is allowed.
-//
-// Return Value:
-//   - nil: indicates that the authorization check passed and the operation is permitted.
-//   - non-nil error: indicates that the authorization failed (either due to insufficient permissions
-//     or an error encountered during the check), the error provides a slice of several failures at once.
+// as defined by the RBAC policy. It decodes the manifest, constructs authorization records,
+// performs attribute checks, and then uses escalationChecker (which delegates to upstream logic)
+// to verify that no privilege escalation is occurring.
 func (a *rbacPreAuthorizer) PreAuthorize(ctx context.Context, manifestManager user.Info, manifestReader io.Reader) ([]ScopedPolicyRules, error) {
-	var allMissingPolicyRules = []ScopedPolicyRules{}
+	var allMissingPolicyRules []ScopedPolicyRules
 	dm, err := a.decodeManifest(manifestReader)
 	if err != nil {
 		return nil, err
 	}
 	attributesRecords := dm.asAuthorizationAttributesRecordsForUser(manifestManager)
 
-	var preAuthEvaluationErrors []error
+	// Compute missing rules using your original logic.
 	missingRules, err := a.authorizeAttributesRecords(ctx, attributesRecords)
 	if err != nil {
-		preAuthEvaluationErrors = append(preAuthEvaluationErrors, err)
+		// If there are errors here, you might choose to log them or collect them.
+		// We'll still try to run the upstream escalation check.
```

Comment:
We still need to collect and return these errors. This is the equivalent of the SelfSubjectAccessReview. That the check itself fails to even tell us whether we had missing rules, we need to tell the user what happened.

Reply:
this section of code was pulled back out in a reverting commit, so marking the comment resolved